src/components/MemeCardComponent.js

```diff
@@ -1,4 +1,7 @@
 import { useDispatch, useSelector } from "react-redux";
+import Card from 'react-bootstrap/Card';
+import Button from 'react-bootstrap/Button';
+
 
 export default function MemeCard(props) {
     const memeId = props.meme.id;
```

Comment:
you used `useSelector(state => state[memeId].upvotes)`  its ok in your app, but if you will have new page (new routing) for one meme, or real data from API, its possible that `state[memeId]` doesnt exist, your app will crash. 

you can use ` const meme = useSelector(state => state[memeId])`
and then 
```
if(!meme){ return null } // or redirect to err page

// and use in code eg: meme.upvotes
```


By the way, You have all object `meme` in props,  you dont have to get it from reducer :)